src/node/txreconciliation.cpp

```diff
@@ -216,6 +233,43 @@ class TxReconciliationTracker::Impl
         recon_state.m_local_set.erase(wtxid_to_remove);
     }
 
+    std::optional<std::pair<uint16_t, uint16_t>> MaybeRequestReconciliation(NodeId peer_id) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        if (!IsPeerRegistered(peer_id)) return std::nullopt;
+        auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+
+        if (!recon_state.m_we_initiate) return std::nullopt;
+
+        if (!m_queue.empty()) {
+            // Request transaction reconciliation periodically to efficiently exchange transactions.
+            // To make reconciliation predictable and efficient, we reconcile with peers in order
+            // based on the queue, taking a delay between requests.
+            auto current_time = GetTime<std::chrono::seconds>();
+            if (m_next_recon_request <= current_time && m_queue.front() == peer_id) {
```

Comment:
I think it might be possible for a peer to abuse this logic and prevent us from reconciling with other peers: If an attacker sends a `SENDTXRCNCL` but stalls the `VERACK`, they would be registered and added to `m_queue` - but `SendMessages()` will never get to the part where `REQTXRCNCL` is sent for this peer because we abort early [here](https://github.com/bitcoin/bitcoin/blob/0596aa40f77a630d8a21035856fa5fd6838b292e/src/net_processing.cpp#L5367) before hanshake completion. Therefore, once that peer is in the front of the queue, it won't get cleared and prevents us from reconciling with other peers while connected (up to `DEFAULT_PEER_CONNECT_TIMEOUT = 60` seconds).

Reply:
I don't see that yet. No matter what happens to this peer, it will be moved to the end of the queue, and the next peer will be requested in couple seconds.

Reply:
ahhh good point, should be addressed indeed.

Reply:
This is a dangerous piece of code... This could also happen if the peer enters IBD again or something.

Reply:
Yeah - just to bring this alternative up, if we uncoupled peers and had every peer on its separate timer (instead of a queue), would that lead to major efficiency losses?

Reply:
I started with having two separate calls — one for queue management, the other for sending REQTXRCNCL.
I'll submit that shortly. Then I'll think about your idea.

Reply:
I made this code extra safe:
- it's very early in `SendMessages`;
- it's called even if in IBD;

Also reminding that this only applies to outbound. 

Having separate queues for every peer would nullify all our previous tuning and performance benchmarks (both mine and from others), that's why I'd rather stick to the current approach.

Reply:
After all, I decided to introduce independent timers, because my previous idea became too ugly. I think the way i implemented it it doesn't nullify previous benchmarks, and should be robust to manipulations.